### PR TITLE
Fix EnvPool dummy transitions in PPO GAE

### LINKED ISSUE #562: GAE Bug for Envpool: Dummy Step Leak
Hello,

I've noticed a bug related to all PPO Atari scripts that use envpool. 

Envpool has an odd handling of terminal states. The next transition after done=True is a "dummy" transition from the terminal state to the start state. That is, the stream of experience is:

State: $$S_{T-1} \rightarrow{} S_T \rightarrow{} S_0 \rightarrow{} S_1$$
Done:  `T -> F -> F -> F`

The "Dummy" Step is from $$S_T\rightarrow{} S_0$$. The GAE calculation will set $$\delta = \gamma * V(S_0) * 1 - V_S(T)$$ for the dummy step. Since $$S_T$$ appears as an observation, the network is trained on this incorrect advantage.

## Reproduction

Here is a notebook that visualizes breakout transitions, showing the dummy.

[termination_error_reproduce.ipynb](https://github.com/user-attachments/files/29385604/termination_error_reproduce.ipynb)
[termination_error_reproduce.pdf](https://github.com/user-attachments/files/29385605/termination_error_reproduce.pdf)

## Fix:
In my own code, I handled this by making a wrapper around cleanRL that flags a transition as a dummy transition whether done=true on the last step. Then, the trace is cut (i.e. next value is zero-d out) whenever done=True or dummy=True:

```
def calculate_gaeE(traj_batch, γ, λ,):
    done = traj_batch.done
    is_dummy = traj_batch.info.get('is_dummy', jnp.zeros_like(done))
    cut_e_trace = done | is_dummy
    cut_e_mult = 1.0 - cut_e_trace.astype(jnp.float32)

    scan_inputs = (traj_batch, cut_e_mult)

    def _get_advantages(gae, inputs):
        transition, continue_e = inputs
        
        # --- Extrinsic ---
        delta = transition.reward + γ * transition.next_value * continue_e - transition.value
        gae = delta + (γ * λ * continue_e * gae)
        
        return gae, gae

    initial_acc = jnp.zeros_like(traj_batch.value[0])
    
    _, advantages = jax.lax.scan(
        _get_advantages, initial_acc, scan_inputs, reverse=True, unroll=16
    )
    
    return advantages, advantages + traj_batch.value
```

### PR DESCRIPTION
**Pull request by lindicaphxag-tech**
## Summary

Fixes #562.

EnvPool can emit a dummy transition from a terminal observation back to the reset observation with done=False. PPO GAE now cuts the trace when either side of an EnvPool transition is terminal. The correction is applied consistently to Torch PPO, RND PPO, JAX PPO, and JAX scan PPO.

## Validation

- python -m pytest tests/test_envpool_gae.py -q — 1 passed.
- python -m black --check --line-length=127 cleanrl/ppo_atari_envpool.py cleanrl/ppo_rnd_envpool.py cleanrl/ppo_atari_envpool_xla_jax.py cleanrl/ppo_atari_envpool_xla_jax_scan.py tests/test_envpool_gae.py tests/test_jax_compute_gae.py — passed.
- Python compile checks for all changed files — passed.
- The existing JAX test suite requires optional flax/JAX dependencies not installed in this local Python environment.

## Duplicate-work check

Issue #562 is still open. No open PR references #562; the similarly named termination/truncation PR #448 changes a different continuous-action example.

## AI assistance

AI assistance was used to prepare this draft. The human submitter must review and understand every changed line and run the project's EnvPool/JAX CI tests before marking this PR ready for review.

### CODE REVIEW

**Comment by lindicaphxag-tech** [comment]
CI note: the failing jobs currently stop during dependency installation before running this change. uv reports that the existing project pin jaxlib==0.4.7 has no resolvable version; this is present in the base branch configuration and is unrelated to the patch. The repository's build/pre-commit and several non-JAX jobs pass.

**Comment by lindicaphxag-tech** [comment]
Closing this draft in favor of the higher-confidence Stable-Baselines3 PR #2287. CleanRL’s CI is blocked by the base branch’s unresolved jaxlib==0.4.7 dependency (tracked separately in issue #540), so this draft is not a suitable completed submission under the requested CI-green standard.

**Comment by lindicaphxag-tech** [comment]
Closing this draft in favor of the higher-confidence Stable-Baselines3 PR #2287. CleanRL’s CI is blocked by the base branch’s unresolved jaxlib==0.4.7 dependency (tracked separately in issue #540), so this draft is not a suitable completed submission under the requested CI-green standard.
